.github/workflows/clp-core-build.yaml

```
@@ -201,17 +208,37 @@ jobs:
         run: "chown $(id -u):$(id -g) -R ."
         shell: "bash"
 
-      - uses: "./.github/actions/clp-core-build"
-        env:
-          OS_NAME: "ubuntu-jammy"
+      - uses: "./.github/actions/clp-core-image-run"
         with:
           image_name: "${{env.DEPS_IMAGE_NAME_PREFIX}}${{env.OS_NAME}}"
-          use_shared_libs: "${{matrix.use_shared_libs}}"
           use_published_image: >-
             ${{needs.filter-relevant-changes.outputs.ubuntu_jammy_image_changed == 'false'
             || (github.event_name != 'pull_request' && github.ref == 'refs/heads/main')}}
-          upload_binaries: "${{matrix.upload_binaries}}"
-          binaries_artifact_name: "${{env.BINARIES_ARTIFACT_NAME_PREFIX}}${{env.OS_NAME}}"
+          run_command: >-
+            task deps:core &&
+            python3 /mnt/clp/components/core/tools/scripts/utils/build-and-run-unit-tests.py
+            ${{matrix.use_shared_libs == 'true' && '--use-shared-libs' || ''}}
+            --source-dir /mnt/clp/components/core
+            --build-dir /mnt/clp/components/core/build
+            --num-jobs $(getconf _NPROCESSORS_ONLN)
+
+      - if: "${{matrix.upload_binaries}}"
```

```suggestion
      - if: "matrix.upload_binaries == 'true'"
```

I think this is more rigorous since otherwise, GitHub will coerce truthy values to true (e.g., the typo "fals" would evaluate to true).